stockflux-securities-master/src/components/InputForm.js

```diff
@@ -188,32 +203,31 @@ const InputForm = ({ match }) => {
           </div>
           <div className="input-buttons-container">
             <Button
-              size="large"
+              size={ButtonSize.LARGE}
               text={match.params.securityId ? "Save" : "Create"}
-              className={
-                (state === InputFormState.SENDING ? "in-progress " : "") +
-                "input-submit-button"
-              }
+              className={`input-submit-button ${state.fetchStatus ===
+                InputFormState.SENDING && " in-progress"}`}
             />
             {match.params.securityId && (
               <Confirmation confirmationText="Are you sure you want to delete this security?">
                 <Button
                   type="button"
                   className="input-delete-button"
                   text="Delete"
-                  size="large"
+                  size={ButtonSize.LARGE}
                   onClick={deleteSecurity}
                 />
               </Confirmation>
             )}
           </div>
         </form>
       )}
-      {!!messages &&
-        (state === InputFormState.ERROR ||
-          state === InputFormState.SUCCESS) && (
-          <Alert type={state} messages={messages} />
-        )}
+      {!!state.messages && (
```

Comment:
Assuming `state.messages` is an array here, `!!state.messages` will be `true` when the array is empty here, resulting in rendering the alert even when there aren't any messages to show. Remember our discussion about trying to be more explicit and careful with types and not relying on JavaScript coercing them? I think you should change this to:

    {state.messages.length > 0 && (

Reply:
Ah I guess you're right. So it hasn't been a problem so far because it's just been rendering an empty string but still best to avoid rendering when it's not needed